manifests/params.pp

```diff
@@ -179,13 +192,16 @@
 
   # Passenger config
   $server_app_root = "${dir}/rack"
-  $server_ssl_dir  = "${vardir}/ssl"
+  $server_ssl_dir  = $ssldir
 
   $server_package     = undef
   $server_version     = undef
-  $client_package     = $::osfamily ? {
-    'Debian' => ['puppet-common','puppet'],
-    default  => ['puppet'],
+  $client_package     = versioncmp($::puppetversion, '4.0') ? {
+    '-1'     => $::osfamily ? {
+      'debian' => ['puppet-common','puppet'],
+      default   => ['puppet'],
+    },
+    default => 'puppet-agent',
```

Comment:
Ah, damn... that whole logic would indeed not find it's place in params.pp anymore... yes, let's go for a variable in params.pp then. :heavy_dollar_sign: